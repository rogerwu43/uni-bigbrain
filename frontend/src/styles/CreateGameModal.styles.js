import styled from 'styled-components';

export const CreateGameModal = styled.div`
  background: white;
  display: ${props => props.active ? 'block' : 'none'};
  padding: 10px 10px 30px;
  position: fixed;
  height: auto;
  left:50%;
  max-width: 900px;
  top:50%;
  transform: translate(-50%,-50%);
  width: 90%;
`;

export const ModalBackground = styled.div`
  background: rgba(0, 0, 0, 0.6);
  display: ${props => props.active ? 'block' : 'none'};position: fixed;
  height: 100%;
  left: 0;
  top: 0;
  width:100%;
`;

export const FormError = styled.p`
  color: red;
`;
