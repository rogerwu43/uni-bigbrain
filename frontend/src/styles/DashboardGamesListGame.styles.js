import styled from 'styled-components';

export const DashboardGamesListGame = styled.div`
  background-color: white;
  display: flex;
  flex-direction: column;
  height: 150px;
  margin: 10px 0px 10px;
`;

export const Data = styled.div`
  display: flex;
  flex: 1;
  flex-direction: column;
  overflow: hidden;
`;

export const Details = styled.div`
  display: flex;
  height: 100px;
  padding: 5px;
`;

export const Options = styled.div`
  align-items: center;
  display: flex;
  height: 50px;
  justify-content: flex-start;
  padding: 0px 5px 5px;
`;

export const SubDetails = styled.div`
  align-items: center;
  background-color: #F2F2F2;
  display: flex;
  height: 50px;
  justify-content: flex-start;

  & > p {
    background-color: #6E6E6E;
    border-radius: 14px;
    color: white;
    font-size: 12px;
    margin: 0 15px;
    padding: 1px 10px 2px;
  }
`;

export const Thumbnail = styled.div`
  align-items: flex-end;
  background-color: #CCCCCC;
  background-image: url(${props => props.img});
  background-position: center;
  background-repeat: no-repeat;
  background-size: 100px;
  display: flex;
  height: 100px;
  justify-content: center;
  margin: 0px 5px 0px 0px;
  min-height: 100px;
  min-width: 100px;
  width: 100px;

  & > p {
    background-color: #000000;
    color: white;
    font-size: 14px;
    margin: 0px;
    padding: 5px 0px;
    width: 100%;
  }
`;

export const DefaultThumbnail = styled(Thumbnail)`
  background-size: 50px;
`;

export const Title = styled.div`
  align-items: center;
  display: flex;
  height: 50px;
  margin: 0px 0px 5px 0px;

  & > p {
    font-size: 20px;
    font-weight: 600;
    margin: 0px;
    overflow: hidden;
    padding: 0px 20px 0px 10px;
    text-overflow: ellipsis;
    white-space: nowrap;
  }
`;
