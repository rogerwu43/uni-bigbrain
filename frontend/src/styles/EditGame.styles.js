import styled from 'styled-components';

export const EditGameQuestion = styled.div`
  align-items: center;
  background-color: #FFFFFF;
  display: flex;
  height: 44px;
  margin: 5px 0px;
  overflow: hidden;

  & > div {
    padding: 0px 6px;
  }

  & > p {
    flex: 1;
    font-weight: 600;
    overflow: hidden;
    padding-left: 15px;
    text-align: left;
    text-overflow: ellipsis;
    white-space: nowrap;
  }
`;

export const EditName = styled.i`
  font-size: 20px;
  padding: 5px;

  &:hover {
    background-color: #C1C1C1;
    border-radius: 10px;
    cursor: pointer;
    opacity: 0.6;
  }
`;

export const GameThumbnail = styled.img`
  display: block;
  margin: auto;
  margin-bottom: 10px;
  max-height: 50px;
  max-width: 50px;
`;

export const ThumbnailForm = styled.form`
  margin: 20px 10px;
  text-align: left;

  & > button {
    display: block;
  }

  & > img {
    margin-bottom: 10px;
    max-height: 100px;
    max-width: 100px;
  }

  & > input {
    display: block;
    margin-bottom: 15px;
  }

  & > p {
    margin: 0px 0px 10px;
  }
`;

export const TitleForm = styled.div`
  align-items: center;
  display: flex;
  flex-direction: column;
  margin: auto;
  width: 100%;
`;

export const TitleInput = styled.input`
  border: 1px solid lightgrey;
  border-radius: 5px;
  font-family: Quicksand, sans-serif;
  font-size: 14px;
  font-weight: 400;
  padding: 10px;
  text-align: center;
  width: 100%;
`;
