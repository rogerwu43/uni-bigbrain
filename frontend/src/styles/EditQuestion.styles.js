import styled from 'styled-components';

// Divs
export const AnswerDiv = styled.div`
  align-items: center;
  display: flex;
  justify-content: center;
  width: 100%;
;`

export const AnswerOptionList = styled.div`
  align-items: center;
  display: flex;
  flex-direction: column;
  justify-content: center;
  width: 80%;
`;

export const CurrentImage = styled.div`
  align-items: center;
  border: 1px dashed grey;
  display: flex;
  flex-direction: column;
  margin-bottom: 10px;
  padding: 10px;

  button + button {
    margin-top: 10px;
  }

  & > img {
    max-height: 200px;
    max-width: 200px;
  }

  & > input {
    font-family: Quicksand, sans-serif;
    margin-bottom: 10px;
  }
`;

export const CurrentVideo = styled.div`
  align-items: center;
  border: 1px dashed grey;
  display: flex;
  flex-direction: column;
  margin-bottom: 10px;
  padding: 10px;

  button + button {
    margin-top: 10px;
  }

  & > iframe {
    margin-bottom: 10px;
    max-height: 200px;
    max-width: 200px;
  }

  & > img {
    margin-bottom: 10px;
    max-height: 200px;
    max-width: 200px;
  }

  & > input {
    font-family: Quicksand, sans-serif;
    height: 24px;
    margin-bottom: 10px;
    width: 80%;
  }
`;

export const HorizontalDiv = styled.div`
  border-bottom: 1px solid lightgrey;
  margin: 10px auto;
  width: 80%;
`;

export const InputOptions = styled.div`
  align-items: center;
  display: flex;
  flex-direction: column;
  margin-bottom: 20px;
`;

export const MediaOptions = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  margin-bottom: 20px;
  width: 80%;
`;

export const StatOptions = styled.div`
  align-items: center;
  display: flex;
  flex-direction: column;

  & > p {
    font-size: 14px;
    font-weight: 600;
    margin: 0px 0px 5px 0px;
  }

  & > select {
    border: none;
    font-family: Quicksand, sans-serif;
    font-size: 12px;
    padding: 10px;
    width: 200px;
  }

  p + select {
    margin-bottom: 20px;
  }
`;

// buttons
export const OptionsButton = styled.button`
  border: none;
  border-radius: 5px;
  color: white;
  display: inline-block;
  font-family: 'Quicksand', sans-serif;
  font-size: 14px;
  font-weight: 600;
  padding: 5px 15px 6px 12px;
  text-align: center;
  text-decoration: none;

  &:hover {
    cursor: pointer;
  }

  & > i {
    margin-right: 10px;
  }

  & + & {
    margin-bottom: 10px;
  }
`;

export const DiscardButton = styled(OptionsButton)`
  background-color: #D72619;
  font-size: 12px;

  &:hover {
    background-color: #E93C30;
  }
`;

export const SaveButton = styled(OptionsButton)`
  background-color: #4CAF50;
  font-size: 24px;
  margin-bottom: 10px;
  padding: 20px 50px;

  &:hover {
    background-color: #5DC661;
  }
`;

// Forms
export const EditQuestion = styled.form`
  margin: auto;
  max-width: 960px;
  padding: 10px 0px;

  @media (max-width: 980px) {
    margin: 0px 10px;
  }
`;

// inputs
export const AnswerInput = styled.input`
  border: 1px solid lightgrey;
  border-radius: 5px;
  font-family: Quicksand, sans-serif;
  font-size: 14px;
  font-weight: 400;
  margin-bottom: 10px;
  padding: 10px;
  text-align: center;
  width: 80%;
`;

export const QuestionInput = styled.input`
  border: 1px solid lightgrey;
  border-radius: 5px;
  font-family: Quicksand, sans-serif;
  font-size: 20px;
  font-weight: 600;
  margin-bottom: 20px;
  padding: 10px;
  text-align: center;
  width: 80%;
`;

// p
export const FormError = styled.p`
  color: red;
  font-style: italic;
  text-align: center;
`;

export const Select = styled.select`
  background-color: white;
`;
