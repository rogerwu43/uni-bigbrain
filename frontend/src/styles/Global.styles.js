import { Link } from 'react-router-dom';
import styled from 'styled-components';

import lobbyBackground from '../images/lobby-background.png';

// Parents (Level 0).

export const ErrorText = styled.p`
  color: red;
  font-size: 14px;
  font-style: italic;
  font-weight: 500px;
`;

export const HorizontalDiv = styled.div`
  border-bottom: 1px solid lightgrey;
  margin: 10px auto;
  width: 100%;
`;

export const LargeCreateButton = styled.button`
  background-color: #0A43B5;
  border: none;
  border-radius: 30px;
  color: white;
  display: inline-block;
  font-family: 'Quicksand', sans-serif;
  font-size: 20px;
  font-weight: 700;
  margin: 20px 0;
  padding: 10px 21px 11px 19px;
  text-align: center;
  text-decoration: none;

  &:focus {
    outline: none;
  }

  &:hover {
    background-color: #285ECA;
    cursor: pointer;
  }

  & > i {
    margin-right: 10px;
  }
`;

export const MediumButton = styled.button`
  border: none;
  border-radius: 5px;
  color: #7A7A7A;
  display: inline-block;
  font-family: 'Quicksand', sans-serif;
  font-size: 11px;
  font-weight: 600;
  height: 30px;
  margin: 0 10px;
  padding: 5px 12px 6px;
  text-align: center;
  text-decoration: none;

  &:focus {
    outline: none;
  }

  &:hover {
    cursor: pointer;
  }

  & > i {
    font-size: 12px;
    margin-right: 10px;
  }
`;

export const MiniLink = styled(Link)`
  color: blue;
  font-size: 14px;

  &:visited {
    color: blue;
  }
`;

export const PageTitle = styled.h1`
  border-color: grey;
  border-style: solid;
  border-width: 1px 0px;
  display: inline-block;
  font-weight: 700;
  padding: 10px 30px;
`;

export const PageWrapper = styled.div`
  background-image: url(${props => props.background ? lobbyBackground : ''});
  height: calc(100% - 50px);
  margin: auto;
  max-width: 960px;
  text-align: center;

  @media (max-width: 980px) {
    margin: 0 10px;
  }
`;

export const SectionTitle = styled.h2`
  font-weight: 700;
  margin: 20px 0px;
`;

export const SubSectionTitle = styled.h3`
  font-weight: 600;
  margin: 20px 0;
`;

export const SubText = styled.p`
  font-size: 12px;
  font-weight: 400;
`;

// Inheritors (Level 1).

export const BlueMediumButton = styled(MediumButton)`
  background-color: #3D7DD0;
  color: white;
  margin: 0 5px;

  &:hover {
    background-color: #3E8FF3;
  }
`;

export const DarkGreenMediumButton = styled(MediumButton)`
  background-color: #5DAA80;
  color: white;
  margin: 0 5px;

  &:hover {
    cursor: default;
  }
`;

export const GreenMediumButton = styled(MediumButton)`
  background-color: #4CAF50;
  color: white;
  margin: 0 5px;

  &:hover {
    background-color: #5DC661;
  }
`;

export const OrangeMediumButton = styled(MediumButton)`
  background-color: #EB3115;
  color: white;
  margin: 0 5px;

  &:hover {
    background-color: #FF3B1C;
  }
`;

export const RedMediumButton = styled(MediumButton)`
  background-color: #D72619;
  color: white;
  margin: 0 5px;

  &:hover {
    background-color: #E93C30;
  }
`;
