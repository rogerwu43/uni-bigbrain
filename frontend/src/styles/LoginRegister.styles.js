import styled from 'styled-components';

export const LabelInput = styled.div`
  margin-bottom: 15px;
  text-align: left;

  & > label {
    font-size: 14px;
    font-weight: 500;
  }
`;

export const LoginDiv = styled.div`
  align-items: center;
  display: flex;
  flex-direction: column;
`;

export const LoginForm = styled.form`
  align-items: center;
  display: flex;
  flex-direction: column;
`;
