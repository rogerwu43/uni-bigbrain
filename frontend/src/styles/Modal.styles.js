import styled from 'styled-components';

export const Modal = styled.div`
  background: white;
  display: ${props => props.active ? 'block' : 'none'};
  padding: 10px 10px 30px;
  position: fixed;
  height: auto;
  left:50%;
  max-width: 900px;
  top:50%;
  transform: translate(-50%,-50%);
  width: 80%;
  max-height: 70%;
  overflow-y: auto;
`;

export const ModalContent = styled.div`
  overflow-y: initial;
  background-color: #fefefe;
  width: 100%;
  text-align: center;
`;

export const ModalHeader = styled.div`
  font-size: 20px;
  text-align: left;
  margin: 10px;
`;

export const ModalBackground = styled.div`
  background: rgba(0, 0, 0, 0.6);
  display: ${props => props.active ? 'block' : 'none'};position: fixed;
  height: 100%;
  left: 0;
  top: 0;
  width:100%;
`;

export const ModalClose = styled.div`
  color: #aaaaaa;
  float: right;
  font-size: 20px;
  font-weight: bold;

  &:hover {
    cursor: pointer;
  }
`;
