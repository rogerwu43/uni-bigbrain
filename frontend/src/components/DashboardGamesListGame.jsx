import PropTypes from 'prop-types';
import React from 'react';

import defaultThumbnail from '../images/thumbnail-default.svg';
import * as S from '../styles/DashboardGamesListGame.styles';
import * as GS from '../styles/Global.styles';
import api from '../util/Api';
import history from '../util/History';
import ResultsModal from './ResultsModal'
import Confirmation from './Confirmation'

export default function DashboardGamesListGame ({ getGameList, id, showPopup, userToken }) {
  const [data, setData] = React.useState({ active: null, questions: [] });
  const [status, setStatus] = React.useState({ position: -1, questions: [], players: [] });
  const [pastResults, setPastResults] = React.useState([]);
  const [showPastResultsModal, setShowPastResultsModal] = React.useState(false);
  const [lastSessionId, setLastSessionId] = React.useState(0);
  const [showConfirmation, setShowConfirmation] = React.useState(false);
  const [timer, setTimer] = React.useState(0);
  const [questionEnds, setQuestionEnds] = React.useState(-1);

  async function deleteGame () {
    const response = await api.deleteAdminQuizQuizid(id, userToken);
    if (response.status !== 200) {
      const error = await response.json();
      showPopup('Error: ' + error.error);
    }
    getGameList();
  }

  async function startGame () {
    const response = await api.postAdminQuizQuizidStart(id, userToken);
    if (response.status !== 200) {
      const error = await response.json();
      showPopup('Error: ' + error.error);
    }
    getData();
  }

  async function advanceGame () {
    const response = await api.postAdminQuizQuizidAdvance(id, userToken);
    if (response.status !== 200) {
      const error = await response.json();
      showPopup('Error: ' + error.error);
    }
    if (status.position === data.questions.length - 1) {
      setLastSessionId(data.active);
      setShowConfirmation(true);
    }
    getData();
  }

  async function endGame () {
    const response = await api.postAdminQuizQuizidEnd(id, userToken);
    if (response.status !== 200) {
      const error = await response.json();
      showPopup('Error: ' + error.error);
    } else {
      setLastSessionId(data.active);
      setShowConfirmation(true);
    }
    getData();
  }

  function showLastResult () {
    history.push('/results/' + lastSessionId);
  }

  function closeConfirmation () {
    setShowConfirmation(false);
  }

  function showPastSessions () {
    setPastResults(data.oldSessions);
    setShowPastResultsModal(true);
  }

  function closeModal () {
    setShowPastResultsModal(false);
  }

  async function getData () {
    const response = await api.getAdminQuizQuizid(id, userToken);
    const responseData = await response.json();
    if (response.status !== 200) {
      showPopup('Error: ' + responseData.error);
    }
    if (responseData.active !== null) {
      const statusResponse = await api.getAdminSessionSessionidStatus(responseData.active, userToken);
      const statusResponseData = await statusResponse.json();
      if (statusResponse.status !== 200) {
        const error = statusResponseData;
        showPopup('Error: ' + error.error);
      }
      setStatus(statusResponseData.results);
      if (statusResponseData.results.position > -1) {
        const questionStarted = Date.parse(statusResponseData.results.isoTimeLastQuestionStarted);
        const questionEnds = questionStarted + (statusResponseData.results.questions[statusResponseData.results.position].timeLimit) * 1000;
        setQuestionEnds(questionEnds);
        const timeLeft = Math.floor((questionEnds - Date.now()) / 1000);
        if (timeLeft > 0) {
          setTimer(timeLeft);
        }
      }
    }
    setData(responseData);
  }

  React.useEffect(() => {
    getData();
  }, []);

  React.useEffect(() => {
    const timerInterval = timer > 0 && setInterval(() => {
      setTimer(Math.floor((questionEnds - Date.now()) / 1000));
    }, 1000);
    return () => clearInterval(timerInterval);
  }, [timer]);

  function editGame () {
    history.push('/editgame/' + id);
  }

  return (
    <S.DashboardGamesListGame>
      <S.Details>
        {data.thumbnail
          ? <>
              <S.Thumbnail img={data.thumbnail}>
              {data.active
                ? <>
                    <p className='sessionid-give'>
                      Session ID
                      <br></br>
                      <b>{data.active}</b>
                    </p>
                  </>
                : null
              }
              </S.Thumbnail>
            </>
          : <S.DefaultThumbnail img={defaultThumbnail}>
              {data.active
                ? <>
                    <p className='sessionid-give'>
                      Session ID
                      <br></br>
                      <b>{data.active}</b>
                    </p>
                  </>
                : null
              }
            </S.DefaultThumbnail>
        }
        <S.Data>
          <S.Title>
            <p className='dashboard-game-title'>{data.name}</p>
          </S.Title>
          <S.SubDetails>
            <p><b>Questions:</b> {data.questions.length}</p>
            <p><b>Total Time:</b> {data.questions.reduce((totalTime, datum) => totalTime + Number(datum.timeLimit), 0)}</p>
          </S.SubDetails>
        </S.Data>
      </S.Details>
      <S.Options>
        {data.active
          ? <>
              {status.position <= -1
                ? <GS.GreenMediumButton type='button' onClick={advanceGame}>Advance (Currently In: Lobby)</GS.GreenMediumButton>
                : <>
                    {timer > 0
                      ? <GS.DarkGreenMediumButton type='button' onClick={advanceGame} disabled>Q{status.position + 1}, {timer}s left</GS.DarkGreenMediumButton>
                      : <GS.GreenMediumButton type='button' onClick={advanceGame}>Advance (Q{status.position + 1}, time out)</GS.GreenMediumButton>
                    }
                  </>
              }
              <GS.BlueMediumButton type='button' onClick={() => { navigator.clipboard.writeText(location.host + '/join?id=' + data.active) }}><i className='fas fa-clipboard-list'></i> Copy Link</GS.BlueMediumButton>
              <GS.RedMediumButton type='button' onClick={endGame}>End</GS.RedMediumButton>
            </>
          : <>
              <GS.GreenMediumButton type='button' onClick={startGame}>Start Game</GS.GreenMediumButton>
              <GS.BlueMediumButton type='button' className='dashboard-game-edit' onClick={editGame}><i className='fas fa-pen'></i>Edit</GS.BlueMediumButton>
              <GS.RedMediumButton type='button' onClick={deleteGame}><i className='far fa-trash-alt'></i>Delete</GS.RedMediumButton>
              <GS.OrangeMediumButton type='button' onClick={showPastSessions}>Past results</GS.OrangeMediumButton>
            </>
        }
      </S.Options>
      <ResultsModal content={pastResults} active={showPastResultsModal} closeModal={closeModal} />
      <Confirmation content="Would you like to view the results for the quiz?" active={showConfirmation} closeModal={closeConfirmation} onConfirm = {showLastResult} />
    </S.DashboardGamesListGame>
  )
}

DashboardGamesListGame.propTypes = {
  getGameList: PropTypes.func,
  id: PropTypes.number,
  userToken: PropTypes.string,
  showPopup: PropTypes.func
}
