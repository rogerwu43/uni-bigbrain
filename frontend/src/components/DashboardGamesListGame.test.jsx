import { mount } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';

import DashboardGamesListGame from './DashboardGamesListGame';
import * as S from '../styles/DashboardGamesListGame.styles';
import * as GS from '../styles/Global.styles';

describe('DashboardGamesListGame', () => {
  const noop = () => {};

  it('has a game title', () => {
    const test = mount(<DashboardGamesListGame
      getGameList={noop}
      id={0}
      userToken=''
      showPopup={noop}
    />);
    expect(test.find(S.Title)).toHaveLength(1);
  });

  it('has interaction buttons', () => {
    const test = mount(<DashboardGamesListGame
      getGameList={noop}
      id={0}
      userToken=''
      showPopup={noop}
    />);
    expect(test.find(GS.GreenMediumButton).first().text()).toBe('Start Game');
    expect(test.find(GS.BlueMediumButton).first().text()).toBe('Edit');
    expect(test.find(GS.RedMediumButton).first().text()).toBe('Delete');
    expect(test.find(GS.OrangeMediumButton).first().text()).toBe('Past results');
  });

  it('renders with minimal props (snapshot', () => {
    const test = renderer.create(<DashboardGamesListGame
                                    getGameList={noop}
                                    id={0}
                                    userToken=''
                                    showPopup={noop}
                                  />).toJSON()
    expect(test).toMatchSnapshot()
  });
});
