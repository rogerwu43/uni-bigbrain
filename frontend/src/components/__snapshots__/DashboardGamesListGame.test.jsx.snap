// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DashboardGamesListGame renders with minimal props (snapshot 1`] = `
<div
  className="sc-bdvvaa wuBVN"
>
  <div
    className="sc-dkPtyc yyxDx"
  >
    <div
      className="sc-jRQAMF sc-gKckTs jYYhuy jdZQTX"
    />
    <div
      className="sc-gsDJrp gnaghU"
    >
      <div
        className="sc-iCfLBT dwihOV"
      >
        <p
          className="dashboard-game-title"
        />
      </div>
      <div
        className="sc-eCImvq fpIaBI"
      >
        <p>
          <b>
            Questions:
          </b>
           
          0
        </p>
        <p>
          <b>
            Total Time:
          </b>
           
          0
        </p>
      </div>
    </div>
  </div>
  <div
    className="sc-hKwCoD doMlYY"
  >
    <button
      className="sc-kDThTU sc-bkkfTU gTtWsO jZCHVE"
      onClick={[Function]}
      type="button"
    >
      Start Game
    </button>
    <button
      className="sc-kDThTU sc-fotPbf gTtWsO vSZrP dashboard-game-edit"
      onClick={[Function]}
      type="button"
    >
      <i
        className="fas fa-pen"
      />
      Edit
    </button>
    <button
      className="sc-kDThTU sc-dJjZJu gTtWsO iafAsW"
      onClick={[Function]}
      type="button"
    >
      <i
        className="far fa-trash-alt"
      />
      Delete
    </button>
    <button
      className="sc-kDThTU sc-ieebsP gTtWsO eQDaBh"
      onClick={[Function]}
      type="button"
    >
      Past results
    </button>
  </div>
  <div
    className="sc-fKVsgm fMMrwf"
  >
    <div>
      <div
        className="sc-hGPAah dwWXcx"
      >
        <div
          className="sc-bBHwJV bCrJAf"
          onClick={[Function]}
        >
          x
        </div>
        <h2
          className="sc-bqiQRQ ePZAFp"
        >
          Past Results
        </h2>
        <div
          className="sc-dlVyqM sc-iwjezw dA-Dchv aLyYo"
        />
      </div>
    </div>
  </div>
  <div
    className="sc-fKVsgm fMMrwf"
  >
    <div
      className="sc-hGPAah dwWXcx"
    >
      <div
        className="sc-dlVyqM dA-Dchv"
      >
        <p>
          Would you like to view the results for the quiz?
        </p>
        <br />
        <button
          className="sc-kDThTU sc-bkkfTU gTtWsO jZCHVE"
          onClick={[Function]}
        >
          Yes
        </button>
        <button
          className="sc-kDThTU sc-dJjZJu gTtWsO iafAsW"
          onClick={[Function]}
        >
          No
        </button>
      </div>
    </div>
  </div>
</div>
`;
